devicetypes/heltun/heltun-hls01-thermostat.src/heltun-hls01-thermostat.groovy

```diff
@@ -0,0 +1,433 @@
+/**
+ *  Copyright 2021 Sarkis Kabrailian
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "HELTUN HLS01 Thermostat", namespace: "HELTUN", author: "Sarkis Kabrailian", cstHandler: true, ocfDeviceType: "oic.d.thermostat") {
+		capability "Energy Meter"
+		capability "Power Meter"
+		capability "Temperature Measurement"
+		capability "Thermostat Heating Setpoint"
+		capability "Thermostat Mode"
+		capability "Thermostat Operating State"
+		capability "Voltage Measurement"    
+		capability "Configuration"
+		capability "Health Check" 
+		capability "Refresh"
+        
+		fingerprint mfr: "0344", prod: "0004", inClusters: "0x5E,0x85,0x59,0x8E,0x55,0x86,0x72,0x5A,0x73,0x87,0x98,0x9F,0x6C,0x32,0x70,0x42,0x40,0x43,0x31,0x81,0x71,0x22,0x7A", deviceJoinName: "HELTUN Thermostat" //model: "000A"
```

Comment:
I tried it, its not working.

The device is not configured correctly

- Device name after inclusion is the name of the device handler "HELTUN HLS01 Thermostat" instead of the devicejoinname "HELTUN Thermostat".
- Device is not shown in the app, but I can see in the the hub settings with a wrong icon

I kept what I did as a comment please check it and let me know what to do

note that "inClusters" works just fine

Reply:
can you just use the mfr/prod/model fields instead?

Reply:
@greens Both devices share the same mfr/prod/model -> #76116, so additional fields are needed here to distinguish those devices (we already talked about it internally some time ago )

Reply:
We've done this before with the Sensative Strips. For those devices, a certain configuration report prompted us to switch DTHs. One of the DTHs just had no fingerprints:

https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zwave-water-temp-light-sensor.src/zwave-water-temp-light-sensor.groovy#L230

If the command class-based fingerprinting isn't working well, this might be an alternative, presuming there's some way to distinguish the devices via other means? Maybe on add do a check of `zwavInfo.cc` for the differentiating command class?

Reply:
@greens setDeviceType("HELTUN HLS01 Switch") is not working
It says: error physicalgraph.exception.NotFoundException: Device type not found, name: HELTUN HLS01 Thermostat @line -1 (k)

Reply:
you will most likely need to pass the optional first argument which is the namespace, in this case "HELTUN"

and both DTHs will of course need to be self-published to that account

Reply:
@greens I just tried it, I was able to change the device type.
However, the icon is not changed, and the capabilities are not shown correctly.
For example, I changed a device that has a "thermostat mode" capability to a device that doesn't have that capability, the icon of mode remained in the app. Is there any way to handle this ? 
![image](https://user-images.githubusercontent.com/92102906/144114782-3b4a653e-a69f-4d75-882e-6f5fbc57b5d0.png)

Reply:
@Sarkis008 I thought this had been fixed. Does exiting the app and re-opening make any difference?

Reply:
@greens no, only the main icon is changed after exiting the app and re-opening it.
The capabilities are the same as in the picture above

Reply:
@Sarkis008 Does force-closing the SmartThings app making any difference? I suspect that the proper changes have occurred but maybe the SmartThings app is still thinking this has a different presentation? There *might* be a service in the backend that has not updated yet, too...

Reply:
@dkirker I tried force closing the app, logging out and re-logging in, uninstalling the app, and reinstalling it, none of them helped. Note that I'm using IOS.

Reply:
@Sarkis008 Alright. Sorry, I thought it had been fixed. 

I've taken a look again at our matching code and it looks like you should be able to just include 2/3 of manufacturer, model, and product type and then if you include the differentiating command classes in the fingerprint as well the matching should score them well enough to select the right DTH.

so:

`fingerprint mfr: , productId: , inClusters: , deviceJoinName: `

Reply:
@greens yes that used to work, however KKlimczukS said "Since this is a z-wave device - instead of inClusters: please use cc: and sec:"
If I use cc and sec The device is not configured correctly:
- Device name after inclusion is the name of the device handler "HELTUN HLS01 Thermostat" instead of the devicejoinname "HELTUN Thermostat".
- Device is not shown in the app, but I can see in the hub settings with a wrong icon

If you agree, I can keep "inClusters" and that will work
please advise

Reply:
it's okay to use `inClusters`

Note that you don't need to include _all_ the command classes, just the ones that are different between the devices

Reply:
@greens the requested change is done, tested, and working.
Can we fasten the publishing process a little bit, all requested changes are done